### DISCUSSION TITLE
upload crt in vector using helm

### DISCUSSION THREAD

**Discussion by francescoblefari** [vectordotdev/vector#14662] [Q&A]
Hi, I've a question.
How I can upload a CRT file in vector deployed in helm?

**Comment by spencergilbert** [comment]
👋 
I'm not sure I understand. Are you asking how to configure Helm to inject a file into the container filesystem, or how to configure [Vector to load it](https://vector.dev/docs/reference/configuration/sources/vector/#tls.crt_file)?

**Reply by francescoblefari** [reply]
the second, becouse my configuration doesn't work.
![image](https://user-images.githubusercontent.com/27406764/193287179-71609bc7-af26-4afd-a6c0-c86aa150efc2.png)

**Reply by spencergilbert** [reply]
What does your Pod's spec look like?

**Reply by francescoblefari** [reply]
I think are this the specs that you mean..



Name:             vector-0
Namespace:        vector
Priority:         0
Service Account:  vector
Node:             ubuntu/192.168.1.66
Start Time:       Fri, 30 Sep 2022 15:31:35 +0200
Labels:           app.kubernetes.io/component=Aggregator
                  app.kubernetes.io/instance=vector
                  app.kubernetes.io/name=vector
                  controller-revision-hash=vector-856b8d56d7
                  statefulset.kubernetes.io/pod-name=vector-0
                  vector.dev/exclude=true
Annotations:      checksum/config: 2b71f41f2b1257e8bfe131ef87d83bcd3a2c9a0a629e9d565da41a31531d9564
                  cni.projectcalico.org/containerID: 06a2cb2eaf6d6add17794392bd1814ba6608770440b845fdf711f2b5ea8996b8
                  cni.projectcalico.org/podIP: 10.1.243.233/32
                  cni.projectcalico.org/podIPs: 10.1.243.233/32
Status:           Running
IP:               10.1.243.233
IPs:
  IP:           10.1.243.233
Controlled By:  StatefulSet/vector
Containers:
  vector:
    Container ID:  containerd://92a240a9b9035c2128938561377958253b4475317c367cb597158e9a160574c9
    Image:         timberio/vector:0.24.0-distroless-libc
    Image ID:      docker.io/timberio/vector@sha256:5bf45ed2166afa77e94a9032c6df6fe890cde5ef41b082a6725e9872b756d345
    Port:          <none>
    Host Port:     <none>
    Args:
      --config-dir
      /etc/vector/
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    78
      Started:      Fri, 30 Sep 2022 16:08:23 +0200
      Finished:     Fri, 30 Sep 2022 16:08:23 +0200
    Ready:          False
    Restart Count:  12
    Environment:    <none>
    Mounts:
      /etc/vector/ from config (ro)
      /home/ubuntu from data (rw)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-bfkpm (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  data:
    Type:       EmptyDir (a temporary directory that shares a pod's lifetime)
    Medium:     
    SizeLimit:  <unset>
  config:
    Type:               Projected (a volume that contains injected data from multiple sources)
    ConfigMapName:      vector
    ConfigMapOptional:  <nil>
  kube-api-access-bfkpm:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                             node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type     Reason   Age                  From     Message
  ----     ------   ----                 ----     -------
  Warning  BackOff  33s (x186 over 40m)  kubelet  Back-off restarting failed container


Name:             vector-zxhtk
Namespace:        vector
Priority:         0
Service Account:  vector
Node:             ubuntu/192.168.1.66
Start Time:       Fri, 30 Sep 2022 16:08:55 +0200
Labels:           app.kubernetes.io/component=Agent
                  app.kubernetes.io/instance=vector
                  app.kubernetes.io/name=vector
                  controller-revision-hash=5d487f5795
                  pod-template-generation=50
                  vector.dev/exclude=true
Annotations:      checksum/config: 4712f72ab82a1eeaa32e88c974d9dad0613764e47c0f470cc24e49b82f6616c8
                  cni.projectcalico.org/containerID: 9d904dded911203f1b01ca3c4f1bbe1205d6af10540258be32d11c2e575f4548
                  cni.projectcalico.org/podIP: 10.1.243.227/32
                  cni.projectcalico.org/podIPs: 10.1.243.227/32
Status:           Running
IP:               10.1.243.227
IPs:
  IP:           10.1.243.227
Controlled By:  DaemonSet/vector
Containers:
  vector:
    Container ID:  containerd://f70524c53fc644588e83ee57feaff43ccefe3a21df4f5dfdaaf5e14e2d96e2dd
    Image:         timberio/vector:0.24.0-distroless-libc
    Image ID:      docker.io/timberio/vector@sha256:5bf45ed2166afa77e94a9032c6df6fe890cde5ef41b082a6725e9872b756d345
    Port:          8088/TCP
    Host Port:     0/TCP
    Args:
      --config-dir
      /etc/vector/
    State:          Waiting
      Reason:       CrashLoopBackOff
    Last State:     Terminated
      Reason:       Error
      Exit Code:    78
      Started:      Fri, 30 Sep 2022 16:11:59 +0200
      Finished:     Fri, 30 Sep 2022 16:11:59 +0200
    Ready:          False
    Restart Count:  5
    Environment:
      VECTOR_SELF_NODE_NAME:       (v1:spec.nodeName)
      VECTOR_SELF_POD_NAME:       vector-zxhtk (v1:metadata.name)
      VECTOR_SELF_POD_NAMESPACE:  vector (v1:metadata.namespace)
      PROCFS_ROOT:                /host/proc
      SYSFS_ROOT:                 /host/sys
    Mounts:
      /etc/vector/ from config (ro)
      /home/ubuntu/Scrivania from data (rw)
      /host/proc from procfs (ro)
      /host/sys from sysfs (ro)
      /var/lib from var-lib (ro)
      /var/log/ from var-log (ro)
      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-j676t (ro)
Conditions:
  Type              Status
  Initialized       True 
  Ready             False 
  ContainersReady   False 
  PodScheduled      True 
Volumes:
  config:
    Type:               Projected (a volume that contains injected data from multiple sources)
    ConfigMapName:      vector
    ConfigMapOptional:  <nil>
  data:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/vector
    HostPathType:  
  var-log:
    Type:          HostPath (bare host directory volume)
    Path:          /var/log/
    HostPathType:  
  var-lib:
    Type:          HostPath (bare host directory volume)
    Path:          /var/lib/
    HostPathType:  
  procfs:
    Type:          HostPath (bare host directory volume)
    Path:          /proc
    HostPathType:  
  sysfs:
    Type:          HostPath (bare host directory volume)
    Path:          /sys
    HostPathType:  
  kube-api-access-j676t:
    Type:                    Projected (a volume that contains injected data from multiple sources)
    TokenExpirationSeconds:  3607
    ConfigMapName:           kube-root-ca.crt
    ConfigMapOptional:       <nil>
    DownwardAPI:             true
QoS Class:                   BestEffort
Node-Selectors:              <none>
Tolerations:                 node.kubernetes.io/disk-pressure:NoSchedule op=Exists
                             node.kubernetes.io/memory-pressure:NoSchedule op=Exists
                             node.kubernetes.io/not-ready:NoExecute op=Exists
                             node.kubernetes.io/pid-pressure:NoSchedule op=Exists
                             node.kubernetes.io/unreachable:NoExecute op=Exists
                             node.kubernetes.io/unschedulable:NoSchedule op=Exists
Events:
  Type     Reason     Age                   From               Message
  ----     ------     ----                  ----               -------
  Normal   Scheduled  3m22s                 default-scheduler  Successfully assigned vector/vector-zxhtk to ubuntu
  Normal   Pulled     111s (x5 over 3m22s)  kubelet            Container image "timberio/vector:0.24.0-distroless-libc" already present on machine
  Normal   Created    111s (x5 over 3m22s)  kubelet            Created container vector
  Normal   Started    111s (x5 over 3m22s)  kubelet            Started container vector
  Warning  BackOff    96s (x10 over 3m21s)  kubelet            Back-off restarting failed container

**Reply by francescoblefari** [reply]
And this is the error:

2022-09-30T14:14:51.798416Z ERROR vector::topology: Configuration error. error=Source "my_source_id": Could not open certificate file "/home/ubuntu/Scrivania/vector_config/rootCA.crt": No such file or directory (os error 2)

**Reply by spencergilbert** [reply]
Sorry for the delay, could you also share your config for Vector?
